addons/l10n_it_edi_withholding/__init__.py

```diff
@@ -0,0 +1,92 @@
+from . import models
+from odoo import api, SUPERUSER_ID
+import logging
+
+_logger = logging.getLogger(__name__)
+
+def _l10n_it_edi_add_accounts(env, company):
+    """ Create the transition accounts """
+    generated_accounts_ref = {}
+    account_templates = {account_code: env.ref(f'l10n_it_edi_withholding.{account_code}') for account_code in ['1611', '2603']}
+    try:
+        with env.cr.savepoint():
+            for account_code, account_template in account_templates.items():
+                template_vals = [(account_template, company.chart_template_id._get_account_vals(company, account_template, account_code, {}))]
+                generated_accounts_ref[account_template] = company.chart_template_id._create_records_with_xmlid('account.account', template_vals, company)
+            _logger.info("Created withholding accounts for company: %s(%s).", company.name, company.id)
+    except psycopg2.errors.UniqueViolation:
+        generated_accounts_ref = {}
+        _logger.error("Cash basis transition accounts already exist for company: %s(%s).", company.name, company.id)
+    return generated_accounts_ref
+
+def _l10n_it_edi_withholding_add_taxes(env, company):
+    """ Create the new taxes on existing company """
+    templates = env['account.tax.template']
+    generated_taxes_ref = {}
+    try:
+        with env.cr.savepoint():
+            for xml_id in (
+                '20awi', '20vwi',
+                '20awc', '20vwc',
+                '23awo', '23vwo',
+                '4vcp', '4acp',
+                '4vinps', '4ainps'
+            ):
+                templates |= env.ref(f"l10n_it_edi_withholding.{xml_id}")
+            generated_taxes_ref = templates._generate_tax(company)
+            _logger.info("Created withholding taxes for company: %s(%s).", company.name, company.id)
+
+            # Increase the sequence number of the old taxes multiplying it by 10 and adding 21
+            # so that the withholding can have sequence=10 and the pension fund sequence=20
+            offset = 21
+            all_taxes = env['account.tax'].search([('company_id', '=', company.id)])
+            for tax in all_taxes.filtered(lambda x: (x.sequence <= 20 and not x.l10n_it_pension_fund_type and not x.l10n_it_withholding_type)):
+                if not tax.l10n_it_withholding_type and not tax.l10n_it_pension_fund_type:
+                    tax.sequence += offset
+            _logger.info("Increased sequence number of old taxes by %s for company: %s(%s).", offset, company.name, company.id)
+    except psycopg2.errors.UniqueViolation:
+        generated_taxes_ref = {}
+        _logger.error("Withholding and Pension Fund taxes already exist for company: %s(%s).", company.name, company.id)
+
+    return generated_taxes_ref
+
+def _l10n_it_edi_setup_accounts_on_taxes(env, company, generated_taxes_ref, generated_accounts_ref):
+    """ Setup the accounts on the taxes, after the accounts have been created """
+    with env.cr.savepoint():
```

Comment:
I've seen it on the other file... Doesn't it do everything or nothing but not only part of it?

Reply:
@lordkrandel If something goes wrong it will go back here, but if something goes wrong it will raise anyways (so the transaction should be rolled back entirely)

Reply:
Copied from https://github.com/odoo/odoo/blob/16.0/addons/l10n_in_tcs_tds/__init__.py#L17 , I will remove. But what's the difference there? isn't it useless there too then?